intranet/apps/printing/views.py

```diff
@@ -27,43 +27,61 @@ class InvalidInputPrintingError(Exception):
     """An error occurred while printing, but it was due to invalid input from the user and is not worthy of a ``CRITICAL`` log message."""
 
 
-def get_printers() -> List[str]:
-    """ Returns the list of available printers.
+def get_printers() -> Dict[str, str]:
+    """Returns a dictionary mapping name:description for available printers.
 
     This requires that a CUPS client be configured on the server.
-    Otherwise, this returns an empty list.
+    Otherwise, this returns an empty dictionary.
 
     Returns:
-        A list of available printers.
+        A dictionary mapping name:description for available printers.
     """
 
     key = "printing:printers"
     cached = cache.get(key)
-    if cached:
+    if cached and isinstance(cached, dict):
         return cached
     else:
         try:
-            output = subprocess.check_output(["lpstat", "-a"], universal_newlines=True, timeout=10)
+            output = subprocess.check_output(["lpstat", "-l", "-p"], universal_newlines=True, timeout=10)
         # Don't die if cups isn't installed.
         except FileNotFoundError:
             return []
-        # Don't die if lpstat -a fails
+        # Don't die if lpstat fails
         except (subprocess.CalledProcessError, subprocess.TimeoutExpired):
             return []
-        lines = output.splitlines()
-        names = []
-        for line in lines:
-            if "requests since" in line:
-                names.append(line.split(" ", 1)[0])
 
-        if "Please_Select_a_Printer" in names:
-            names.remove("Please_Select_a_Printer")
-
-        if "" in names:
-            names.remove("")
-
-        cache.set(key, names, timeout=settings.CACHE_AGE["printers_list"])
-        return names
+        PRINTER_LINE_RE = re.compile(r"^printer\s+(\w+)", re.ASCII)
+        DESCRIPTION_LINE_RE = re.compile(r"^\s+Description:\s+(.*)\s*$", re.ASCII)
+
+        printers = {}
+        last_name = None
+        for line in output.splitlines():
+            match = PRINTER_LINE_RE.match(line)
+            if match is not None:
+                # Pull out the name of the printer
+                name = match.group(1)
+                if name != "Please_Select_a_Printer":
+                    # By default, use the name of the printer instead of the description
+                    printers[name] = name
+                    # Record the name of the printer so when we parse the rest of the
+                    # extended description we know which printer it's referring to.
+                    last_name = name
+            else:
+                # If we've seen a line with the name of a printer before
+                print(last_name)
```

Comment:
Is this left over from debugging? If we want to log, we should use `logging`.

Reply:
Yes, this (and the other `print` statement a few lines down) are left over from debugging. I don't know how I missed them.